sdk/src/main/java/com/hedera/hashgraph/sdk/TransferTransaction.java

```diff
@@ -187,4 +178,66 @@ boolean onFreeze(TransactionBody.Builder bodyBuilder) {
     void onScheduled(SchedulableTransactionBody.Builder scheduled) {
         scheduled.setCryptoTransfer(build());
     }
+
+    private Map<AccountId, Long> getTokenTransferMap(TokenId tokenId) {
+        // Cannot use `Map.merge()` as it uses `BiFunction`
+        var map = tokenTransfers.containsKey(tokenId) ?
+            Objects.requireNonNull(tokenTransfers.get(tokenId)) :
+            new HashMap<AccountId, Long>();
+        tokenTransfers.put(tokenId, map);
+        return map;
+    }
+
+    private List<NftTransfer> getNftTransferList(TokenId tokenId) {
```

Comment:
Can we update this to `Map<TokenId, NftTransfer>` where `NftTransfer` does not contain `tokenId`. Right now, the Go SDK returns `map[TokenID]NftTransfer` which _I thinik_ makes a little more sense. Especially considering the protobufs `TokenTransferList` contains `token` and `nftTransfers` for that specific `TokenId`.

Reply:
Not sure what your intention is, there.  It's possible (and expected) for multiple NFTs with the same `TokenId` to be transferred in one `TransferTransaction`, and it would be an error to prevent that.  Unless you meant `Map<NftId, NftTransfer>` instead of `Map<TokenId, NftTransfer>`?

Reply:
Closing as this was my misunderstanding.